src/main/java/org/sonarlint/intellij/config/global/wizard/OrganizationStep.java

```diff
@@ -60,18 +62,32 @@ public OrganizationStep(WizardModel model) {
         }
       }
     });
-    selectOtherOrganizationButton.addActionListener(e -> {
+    selectOtherOrganizationButton.addActionListener(e -> enterCustomOrganizationKey());
+  }
+
+  private void enterCustomOrganizationKey() {
+    while (true) {
       String organizationKey = Messages.showInputDialog(panel, "Please enter the organization key", "Add Another Organization", null);
       if (StringUtil.isNotEmpty(organizationKey)) {
         boolean found = selectOrganizationIfExists(organizationKey);
         if (!found) {
-          RemoteOrganization newOrg = new Organization(organizationKey);
-          listModel.add(0, newOrg);
-          orgList.setSelectedIndex(0);
-          orgList.ensureIndexIsVisible(0);
+          GetOrganizationTask task = new GetOrganizationTask(model.createServerWithoutOrganization(), organizationKey);
+          ProgressManager.getInstance().run(task);
+          if (task.organization() != null) {
+            listModel.add(0, task.organization());
+            orgList.setSelectedIndex(0);
+            orgList.ensureIndexIsVisible(0);
+          } else if (task.getException() != null) {
+            Messages.showErrorDialog("Failed to fetch organization from SonarQube server: " + task.getException().getMessage(), "Connection Failure");
+            continue;
+          } else {
+            Messages.showErrorDialog(String.format("Organization '%s' not found. Please enter the key of an existing organization.", organizationKey), "Organization Not Found");
+            continue;
+          }
         }
       }
-    });
+      break;
```

Comment:
This may be a matter of taste, I would find it simpler to explicitly `break` when ready. That is, `if (found) break` and reduce the nesting, and then also when `task.organization() != null` then `break`. And the 2 `continue` can be dropped.